yroots/IntervalChecks.py

```diff
@@ -545,365 +544,329 @@ def quadratic_check_3D(test_coeff, intervals, change_sign, tol):
         A list of the results of each interval. False if the function is guarenteed to never be zero
         in the unit box, True otherwise
     """
+    mask = [True]*len(intervals)
+
     if test_coeff.ndim != 3:
-        return [True]*len(intervals)
+        return mask
 
     #Padding is slow, so check the shape instead.
+    c = [0]*10
     shape = test_coeff.shape
-    c0 = test_coeff[0,0,0]
+    c[0] = test_coeff[0,0,0]
     if shape[0] > 1:
-        c1 = test_coeff[1,0,0]
-    else:
-        c1 = 0
+        c[1] = test_coeff[1,0,0]
     if shape[1] > 1:
-        c2 = test_coeff[0,1,0]
-    else:
-        c2 = 0
+        c[2] = test_coeff[0,1,0]
     if shape[2] > 1:
-        c3 = test_coeff[0,0,1]
-    else:
-        c3 = 0
+        c[3] = test_coeff[0,0,1]
     if shape[0] > 1 and shape[1] > 1:
-        c4 = test_coeff[1,1,0]
-    else:
-        c4 = 0
+        c[4] = test_coeff[1,1,0]
     if shape[0] > 1 and shape[2] > 1:
-        c5 = test_coeff[1,0,1]
-    else:
-        c5 = 0
+        c[5] = test_coeff[1,0,1]
     if shape[1] > 1 and shape[2] > 1:
-        c6 = test_coeff[0,1,1]
-    else:
-        c6 = 0
+        c[6] = test_coeff[0,1,1]
     if shape[0] > 2:
-        c7 = test_coeff[2,0,0]
-    else:
-        c7 = 0
+        c[7] = test_coeff[2,0,0]
     if shape[1] > 2:
-        c8 = test_coeff[0,2,0]
-    else:
-        c8 = 0
+        c[8] = test_coeff[0,2,0]
     if shape[2] > 2:
-        c9 = test_coeff[0,0,2]
-    else:
-        c9 = 0
-
-    #function for evaluating c0 + c1x + c2y +c3z + c4xy + c5xz + c6yz + c7x^2 + c8y^2 + c9z^2
-    #(note these are chebyshev monomials)
-    eval_func = lambda x,y,z: c0 + c1*x + c2*y + c3*z + c4*x*y + c5*x*z + c6*y*z +\
-                                    c7*(2*x**2-1) + c8*(2*y**2-1) + c9*(2*z**2-1)
+        c[9] = test_coeff[0,0,2]
 
     #The sum of the absolute values of everything else
-    other_sum = np.sum(np.abs(test_coeff)) - np.sum(np.abs([c0,c1,c2,c3,c4,c5,c6,c7,c8,c9])) + tol
+    other_sum = np.sum(np.abs(test_coeff)) - sum([fabs(coeff) for coeff in c]) + tol
+
+    #function for evaluating c0 + c1x + c2y +c3z + c4xy + c5xz + c6yz + c7T_2(x) + c8T_2(y) + c9T_2(z)
+    # Use the Horner form because it is much faster, also do any repeated computatons in advance
+    k0 = c[0]-c[7]-c[8]-c[9]
+    k7 = 2*c[7]
+    k8 = 2*c[8]
+    k9 = 2*c[9]
+    def eval_func(x,y,z):
+        return k0 + (c[1] + k7 * x + c[4] * y + c[5] * z) * x + \
+                    (c[2] + k8 * y + c[6] * z) * y + \
+                    (c[3] + k9 * z) * z
+
+    #TODO: fine tooth comb to avoid repeated computations
```

Comment:
Do we still want this TODO here, or is this what the pull request is meant to have done?